ga4gh/datamodel/variants.py

```diff
@@ -547,3 +596,162 @@ def buildMetadata(
+        return effect
+
+    def convertSeqOntology(self, seqOntStr):
+        seqOntTerms = seqOntStr.split('&')
+        soTerms = []
+        for soName in seqOntTerms:
+            so = self._createGaOntologyTermSo()
+            so.name = soName
+            if self._sequenceOntology is not None:
+                so.id = self._sequenceOntology.getId(soName)
+            soTerms.append(so)
+            # TODO We must fill the ontology ID based on the SO name
+        return soTerms
+
+    def convertVariantAnnotation(self, record):
+        """
+        Converts the specified record into a GA4GH Annotaiton object.
+        """
+        # First we need the Variant
+        variant = self.convertVariant(record, None)
+        # Create annotaiton
+        annotation = self._createGaVariantAnnotation()
+        annotation.variant = variant
+        annotation.variantId = variant.id
+        if record.id is not None:
+            annotation.coLocatedVariants = record.id.split(';')
+        # Convert annotations from INFO field into TranscriptEffect
+        annStr = record.info.get('ANN')
+        transcriptEffects = []
+        if annStr:
+            for ann in annStr.split(','):
+                transcriptEffects.append(self.convertTranscriptEffect(ann))
+        annotation.transcriptEffects = transcriptEffects
+        annotation.id = self.getVariantAnnotationId(variant, annotation)
+        return annotation
+
+    def getVariantAnnotationId(self, gaVariant, gaAnnotation):
+        md5 = self.hashVariantAnnotation(gaVariant, gaAnnotation)
+        compoundId = datamodel.VariantAnnotationCompoundId(
+            self.getCompoundId(), gaVariant.referenceName,
+            gaVariant.start, md5)
+        return str(compoundId)
+
+    def getVariantId(self, gaVariant):
+        """
+        Returns an ID string suitable for the specified GA Variant Annotation
+        object in this variant set.
+        """
+        md5 = self.hashVariant(gaVariant)
+        compoundId = datamodel.VariantAnnotationCompoundId(
+            self.getCompoundId(), gaVariant.referenceName,
+            gaVariant.start, md5)
+        return str(compoundId)
+
+    def getTranscriptEffectId(self, gaTranscriptEffect):
+        effs = [eff.name for eff in gaTranscriptEffect.effects]
+        return hashlib.md5(
+            gaTranscriptEffect.alternateBases +
+            "\t" + gaTranscriptEffect.featureId +
+            "\t" + str(effs) +
+            "\t" + gaTranscriptEffect.HGVSp +
+            "\t" + gaTranscriptEffect.HGVSc
+            ).hexdigest()
+
+    def hashVariantAnnotation(cls, gaVariant, gaVariantAnnotation):
+        """
+        Produces an MD5 hash of the ga variant and gaVariantAnnotaiton objects
+        """
+        treffs = [treff.id for treff in gaVariantAnnotation.transcriptEffects]
+        return hashlib.md5(
+            gaVariant.referenceBases +
+            str(tuple(gaVariant.alternateBases)) +
```

Comment:
Ditto here for .format. It saves you the trouble of calling ``str``, as it automatically converts to string.